### ISSUE TITLE
GHC sources: AST of input and AST of formatted code differ

### ISSUE THREAD

**Issue by int-index** [tweag/ormolu#200]
There are several files in the GHC sources that trigger this error:

* [`compiler/ghci/Debugger.hs`](https://github.com/ghc/ghc/blob/d2e290d3280841647354ddf5ca9abdd974bce0d5/compiler/ghci/Debugger.hs)
* [`compiler/nativeGen/SPARC/Base.hs`](https://github.com/ghc/ghc/blob/d2e290d3280841647354ddf5ca9abdd974bce0d5/compiler/nativeGen/SPARC/Base.hs)
* [`compiler/utils/Encoding.hs`](https://github.com/ghc/ghc/blob/d2e290d3280841647354ddf5ca9abdd974bce0d5/compiler/utils/Encoding.hs)
* [`compiler/cmm/CmmLive.hs`](https://github.com/ghc/ghc/blob/d2e290d3280841647354ddf5ca9abdd974bce0d5/compiler/cmm/CmmLive.hs)
* [`compiler/cmm/MkGraph.hs`](https://github.com/ghc/ghc/blob/d2e290d3280841647354ddf5ca9abdd974bce0d5/compiler/cmm/MkGraph.hs)
* [`compiler/cmm/CmmProcPoint.hs`](https://github.com/ghc/ghc/blob/d2e290d3280841647354ddf5ca9abdd974bce0d5/compiler/cmm/CmmProcPoint.hs)
* [`compiler/cmm/CmmLayoutStack.hs`](https://github.com/ghc/ghc/blob/d2e290d3280841647354ddf5ca9abdd974bce0d5/compiler/cmm/CmmLayoutStack.hs)
* [`compiler/cmm/CmmBuildInfoTables.hs`](https://github.com/ghc/ghc/blob/d2e290d3280841647354ddf5ca9abdd974bce0d5/compiler/cmm/CmmBuildInfoTables.hs)
* [`compiler/cmm/Hoopl/Dataflow.hs`](https://github.com/ghc/ghc/blob/d2e290d3280841647354ddf5ca9abdd974bce0d5/compiler/cmm/Hoopl/Dataflow.hs)

**Comment by mrkkrp** [comment]
This report is not helpful. We don't know if some of these are already reported or not. You could as well just wait when we deal with the bugs currently reported and then try again.

**Comment by int-index** [comment]
Here's the output of `ormolu` on these files:

```
AST of input and AST of formatted code differ.
Please, consider reporting the bug.
```

I'm not sure how I'm supposed to make this into a useful bug report.

**Comment by mrkkrp** [comment]
It is true that the message can be improved. It didn't stop @neongreen from creating many helpful issues though.

I'm going to keep this open, but we'll probably invest as much effort into resolving it as you invested into reporting it. That is, when everything else is done, we'll go through the list of files and see if there are still problems.

**Comment by int-index** [comment]
Now that some of my fixes have been merged, the only remaining failure is due to #207:

```
compiler/ghci/Debugger.hs
AST of input and AST of formatted code differ.
  at ghci/Debugger.hs:(171,9)-(194,40)
```

Therefore, I'm closing this issue.
